app.go

```diff
@@ -47,29 +49,31 @@ type optionFunc func(*App)
 
 func (f optionFunc) apply(app *App) { f(app) }
 
-// Provide registers constructors with the application's dependency injection
-// container. Constructors provide one or more types, can depend on other
-// types available in the container, and may optionally return an error. For
-// example:
+// Provide registers any number of constructor functions, teaching the
+// application how to instantiate various types. The supplied constructor
+// functions must return one or more objects, may depend on other types
+// available in the application, and may return an error. For example:
 //
-//  // Provides type *C, depends on *A and *B.
+//  // Constructs type *C, depends on *A and *B.
 //  func(*A, *B) *C
 //
-//  // Provides type *C, depends on *A and *B, and indicates failure by
+//  // Constructs type *C, depends on *A and *B, and indicates failure by
 //  // returning an error.
 //  func(*A, *B) (*C, error)
 //
-//  // Provides type *B and *C, depends on *A, and can fail.
+//  // Constructs types *B and *C, depends on *A, and can fail.
 //  func(*A) (*B, *C, error)
 //
-// The order in which constructors are provided doesn't matter. Constructors
-// are called lazily and their results are cached for reuse.
+// The order in which constructors are provided doesn't matter, and passing
+// multiple Provide options appends to the application's collection of
+// constructors. Constructors are called lazily and their results are cached
```

Comment:
`Constructors are called lazily`

I think it's worth to dive into something like `constructors are called when one or more of their return types are needed ...`

Reply:
Done!